memor/session.py

```diff
@@ -154,16 +154,67 @@ def add_message(self,
             self._messages_status.insert(index, status)
         self._date_modified = get_time_utc()
 
-    def remove_message(self, index: int) -> None:
+    def get_message_by_index(self, index: Union[int, slice]) -> Union[Prompt, Response]:
         """
-        Remove a message from the session object.
+        Get a message from the session object by index.
+
+        :param index: index
+        """
+        return self._messages[index]
+
+    def get_message_by_id(self, message_id: str) -> Union[Prompt, Response]:
+        """
+        Get a message from the session object by message id.
+
+        :param message_id: message id
+        """
+        for index, message in enumerate(self._messages):
+            if message.id == message_id:
+                return self.get_message_by_index(index=index)
+
+    def get_message(self, identifier: Union[int, slice, str]) -> Union[Prompt, Response]:
+        """
+        Get a message from the session object.
+
+        :param identifier: message identifier (index or id)
+        """
+        if isinstance(identifier, (int, slice)):
+            return self.get_message_by_index(index=identifier)
+        if isinstance(identifier, str):
+            return self.get_message_by_id(message_id=identifier)
+
+    def remove_message_by_index(self, index: int) -> None:
+        """
+        Remove a message from the session object by index.
 
         :param index: index
         """
         self._messages.pop(index)
         self._messages_status.pop(index)
         self._date_modified = get_time_utc()
 
+    def remove_message_by_id(self, message_id: str) -> None:
+        """
+        Remove a message from the session object by message id.
+
+        :param message_id: message id
+        """
+        for index, message in enumerate(self._messages):
+            if message.id == message_id:
+                self.remove_message_by_index(index=index)
+                break
+
+    def remove_message(self, identifier: Union[int, str]) -> None:
+        """
+        Remove a message from the session object.
+
+        :param identifier: message identifier (index or id)
+        """
+        if isinstance(identifier, int):
+            self.remove_message_by_index(index=identifier)
+        if isinstance(identifier, str):
+            self.remove_message_by_id(message_id=identifier)
```

Comment:
same for the else that raises error

Reply:
https://github.com/openscilab/memor/pull/108/commits/f8b94775b0ffb49bc94f38322c98f0cadc60075d